src/test/java/org/broadinstitute/hellbender/tools/exome/ACNVModellerUnitTest.java

```diff
@@ -119,4 +125,49 @@ public void testMergeSimilarSegments() {
         final List<SimpleInterval> segmentsTruth = SegmentUtils.readIntervalsFromSegmentFile(SEGMENTS_TRUTH_FILE);
         Assert.assertEquals(segmentsResult, segmentsTruth);
     }
+
+    /**
+     * Same as {@link ACNVModellerUnitTest#testMergeSimilarSegments()}, but without intermediate model refitting.
+     */
+    @Test
+    public void testMergeSimilarSegmentsWithoutRefitting() {
+        final boolean doRefit = false;
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        LoggingUtils.setLoggingLevel(Log.LogLevel.INFO);
+
+        final String tempDir = publicTestDir + "similar-segment-test-without-refitting";
+        final File tempDirFile = createTempDir(tempDir);
+
+        //load data (coverages, SNP counts, and segments)
+        final Genome genome = new Genome(COVERAGES_FILE, TUMOR_ALLELIC_COUNTS_FILE, SAMPLE_NAME);
+        final SegmentedModel segmentedModel = new SegmentedModel(SEGMENT_FILE, genome);
+
+        //initial MCMC model fitting performed by ACNVModeller constructor
+        final ACNVModeller modeller =
+                new ACNVModeller(segmentedModel, NUM_SAMPLES, NUM_BURN_IN, NUM_SAMPLES, NUM_BURN_IN, ctx);
+        //check that model is completely fit at construction
+        Assert.assertTrue(modeller.isModelFit());
+        //perform iterations of similar-segment merging until all similar segments are merged
+        int prevNumSegments;
+        for (int numIterations = 1; numIterations <= MAX_SIMILAR_SEGMENT_MERGE_ITERATIONS; numIterations++) {
+            prevNumSegments = modeller.getACNVModeledSegments().size();
+            modeller.performSimilarSegmentMergingIteration(INTERVAL_THRESHOLD, INTERVAL_THRESHOLD, doRefit);
+            if (modeller.getACNVModeledSegments().size() == prevNumSegments) {
+                break;
+            }
+        }
+        //check that model is not completely fit at this point
+        Assert.assertTrue(!modeller.isModelFit());
+
+        //write final model fit to file; this should force model refit
+        final File finalModeledSegmentsFile =
+                new File(tempDirFile.getAbsolutePath() + "/test-" + FINAL_SEG_FILE_TAG + ".seg");
+        modeller.writeACNVModeledSegmentFile(finalModeledSegmentsFile);
+        Assert.assertTrue(modeller.isModelFit());
+
+        //check final number of segments = 31 (ground truth has 30 segments, so we miss one merge by not performing intermediate refits)
+        final int expectedNumberOfSegments = 31;
```

Comment:
Good catch.  Now that I think about it, I'm not sure what is actually a satisfying test here.  Maybe just require that merging without refitting gets to within a certain threshold of the expected number of segments that merging with refitting finds?  In any case, I'll add comments and get rid of the magic number.